### PR TITLE
MAGN-10353 Crash when installing packages to folder

### PR DESCRIPTION
**Pull request by NightHawk-Michael**
### Purpose

Fixed Issues: MAGN-10353 Crash when installing packages to folder.
Solution was to re-create the non-existed folder just at the moment the event occurs. This helps to prevent any modification of the file (delete or change name) just prior to the event occurrence causing the crash. 

### Declarations

Check these if you believe they are true

- [ ] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.

### Reviewers

@ke-yu 

rootdirectorynotexisted
foldernotfound

### CODE REVIEW

File: src/DynamoCoreWpf/Properties/Resources.en-US.resx [hunk 1]
```diff
@@ -2046,4 +2046,7 @@ Next assemblies were loaded several times:
     <value>Error Opening File</value>
     <comment>Nofication Center Title</comment>
   </data>
+  <data name="PackageFolderNotFound" xml:space="preserve">
+    <value>Package Installation Folder Is Not Found.</value>
```
Comment (ke-yu) [inline src/DynamoCoreWpf/Properties/Resources.en-US.resx]: I dont think the first character of each word should be in uppercase

File: src/DynamoCoreWpf/Views/PackageManager/PackageManagerSearchView.xaml.cs [hunk 2]
```diff
@@ -75,6 +77,14 @@ private void OnRequestShowFileDialog(object sender, PackagePathEventArgs e)
             
             e.Cancel = true;
 
+            // Handle for the case, initialPath does not exist.
+            var errorCannotCreateFolder = PathHelper.CreateFolderIfNotExist(initialPath);
+            if (errorCannotCreateFolder != null)
+            {
+                throw new DirectoryNotFoundException(Wpf.Properties.Resources.PackageFolderNotFound);
```
Comment (ke-yu) [inline src/DynamoCoreWpf/Views/PackageManager/PackageManagerSearchView.xaml.cs]: Throw exception or displaying an error dialog?

File: src/DynamoCoreWpf/Properties/Resources.en-US.resx [hunk 3]
```diff
@@ -2046,4 +2046,7 @@ Next assemblies were loaded several times:
     <value>Error Opening File</value>
     <comment>Nofication Center Title</comment>
```
Comment (gregmarr) [inline src/DynamoCoreWpf/Properties/Resources.en-US.resx]: I noticed in the email for this PR that Notification is spelled wrong here.

File: src/DynamoCoreWpf/Views/PackageManager/PackageManagerSearchView.xaml.cs [hunk 4]
```diff
@@ -75,18 +78,29 @@ private void OnRequestShowFileDialog(object sender, PackagePathEventArgs e)
...
+                    e.Cancel = false;
+                    e.Path = dialog.SelectedPath;
+                }
+               
+            }
+            else
             {
-                e.Cancel = false;
-                e.Path = dialog.SelectedPath;
+                System.Windows.Forms.MessageBox.Show(Wpf.Properties.Resources.PackageFolderNotFound, "File Not Found", MessageBoxButtons.OK, MessageBoxIcon.Error);
```
Comment (ke-yu) [inline src/DynamoCoreWpf/Views/PackageManager/PackageManagerSearchView.xaml.cs]: "File Not Found" is hard coded string, should be localized.

**Pull request closed without merging**